primitives/sr25519/keys_test.go

```diff
@@ -197,6 +198,36 @@ func TestSecretKey(t *testing.T) {
 			t.Fatalf("sk == sk2")
 		}
 	})
+	t.Run("Ed25519Bytes", func(t *testing.T) {
+		ed25519Bytes, _ := hex.DecodeString("28b0ae221c6bb06856b287f60d7ea0d98552ea5a16db16956849aa371db3eb51fd190cce74df356432b410bd64682309d6dedb27c76845daf388557cbac3ca34")
```

Comment:
Add a note saying where the test vector is from.

Reply:
If there are multiple vectors, we could also do a for loop over vector slice:
```
testVectors := []testVector{
     {
        ed25519: "28b0ae221c6bb06856b287f60d7ea0d98552ea5a16db16956849aa371db3eb51fd190cce74df356432b410bd64682309d6dedb27c76845daf388557cbac3ca34"),
        sk: "46ebddef8cd9bb167dc30878d7113b7e168e6f0646beffd77d69d39bad76b47a",
        nonce: "fd190cce74df356432b410bd64682309d6dedb27c76845daf388557cbac3ca34",
    }
    ...
}

for _, v := range testVectors {
}
```
We could also test that we get an error if the size is not ok or if we corrupt the scalar.